core/src/main/java/org/mule/runtime/core/internal/streaming/object/factory/AbstractCursorIteratorProviderFactory.java

```diff
@@ -32,11 +32,10 @@
 public abstract class AbstractCursorIteratorProviderFactory implements CursorIteratorProviderFactory {
 
   protected final StreamingManager streamingManager;
-  protected final boolean trackCursorProviderClose;
+  protected final static boolean trackCursorProviderClose = getBoolean(TRACK_CURSOR_PROVIDER_CLOSE_PROPERTY);
```

Comment:
I don't understand how this test can pass. First parametrization runs fine, but for the second one, this value is not updated.

To avoid reflection hackerism, you can add a `setCursorProviderCloseTracking(boolean enabled)` method on the abstract factories (not in the API interface)